google-cloud-core/src/main/java/com/google/cloud/Binding.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.core.InternalApi;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Binding {
+  private String role;
+  private List<String> members;
+  private Condition condition;
+
+  public static class Builder {
```

Comment:
good call I didn't think about using auto-value and I think you're right in that it can be helpful to be fully immutable reducing the number of builders. 

Please correct me but I don't think I should update the Policy class but only Binding and Condition to use Auto-Value.